File: CPP/home_work/BlackJack/card.cpp
#include "card.h"
#include "utility.h"

card::card(const card& o){
	this->_name=o._name;
	this->_suit=o._suit;
}
void card::set(const int &_name,const int &_suit){
	this->_name=(name)_name;
	this->_suit=(suit)_suit;
}
void card::get(const bool& hide){
	char tmp;
	switch(_name) {
		case jocker: tmp='J'; break;
		case jack: tmp='J'; break;
		case queen: tmp='Q'; break;
		case king: tmp='K'; break;
		case ace: tmp='A'; break;
	};
	std::cout<<(char)218<<(char)196<<(char)196<<(char)196<<(char)196<<(char)196;
	std::cout<<(char)196<<(char)196<<(char)196<<(char)196<<(char)191<<"\n";
	if(hide){
		std::cout<<(char)179<<" +++++++ "<<(char)179<<"\n";
		std::cout<<(char)179<<" +-----+ "<<(char)179<<"\n";
		std::cout<<(char)179<<" +-----+ "<<(char)179<<"\n";
		std::cout<<(char)179<<" +-----+ "<<(char)179<<"\n";
		std::cout<<(char)179<<" +-----+ "<<(char)179<<"\n";
		std::cout<<(char)179<<" +-----+ "<<(char)179<<"\n";
		std::cout<<(char)179<<" +++++++ "<<(char)179<<"\n";
	}
	else{
		if(_name==card10) std::cout<<(char)179<<" "<<(char)_suit<<_name<<"     "<<(char)179<<"\n";
		else if(_name<card10) std::cout<<(char)179<<" "<<(char)_suit<<_name<<"      "<<(char)179<<"\n";
		else std::cout<<(char)179<<" "<<(char)_suit<<tmp<<"      "<<(char)179<<"\n";
		std::cout<<(char)179<<"         "<<(char)179<<"\n";
		std::cout<<(char)179<<"         "<<(char)179<<"\n";
		std::cout<<(char)179<<"    "<<(char)_suit<<"    "<<(char)179<<"\n";
		std::cout<<(char)179<<"         "<<(char)179<<"\n";
		std::cout<<(char)179<<"         "<<(char)179<<"\n";
		if(_name==card10) std::cout<<(char)179<<"     "<<_name<<(char)_suit<<" "<<(char)179<<"\n";
		else if(_name<card10) std::cout<<(char)179<<"      "<<_name<<(char)_suit<<" "<<(char)179<<"\n";
		else std::cout<<(char)179<<"      "<<tmp<<(char)_suit<<" "<<(char)179<<"\n";
	}
	std::cout<<(char)192<<(char)196<<(char)196<<(char)196<<(char)196<<(char)196;
	std::cout<<(char)196<<(char)196<<(char)196<<(char)196<<(char)217<<"\n";
}
int card::get_value()const{
	if(_name==ace) return 11;
	else if(_name>card9) return 10;
	else return _name;
}
card& card::operator=(const card& o){
	this->_name=o._name;
	this->_suit=o._suit;

	return *this;
}

File: CSharp/ConsoleApplications/lessson8/practice/practice/Card.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace practice
{
    enum Rank { _6 = 6, _7, _8, _9, _10, Jack, Queen, King, Ace };
    enum Suit { Hearts = 3, Diamonds, Clubs, Spades };

    class Card
    {
        Rank rank;
        Suit suit;
        public Card(Rank rank, Suit suit)
        {
            this.suit = suit;
            this.rank = rank;
        }
        static public bool operator >(Card o1, Card o2) => o1.rank > o2.rank;
        static public bool operator <(Card o1, Card o2) => !(o1 > o2);
        //public override string ToString() => $"{RankToString()}{(char)suit}";
        public void ShowCard(int indentSize=0) //Метод показа карты с указанием отступа от края
        {
            char s = (char)suit;
            string r = RankToString();
            Console.CursorLeft = indentSize;
            Console.WriteLine("┌─────────┐");
            Console.CursorLeft = indentSize;
            Console.WriteLine($"│ {r}      │");
            Console.CursorLeft = indentSize;
            Console.WriteLine("│         │");
            Console.CursorLeft = indentSize;
            Console.WriteLine("│         │");
            Console.CursorLeft = indentSize;
            Console.WriteLine($"│    {s}    │");
            Console.CursorLeft = indentSize;
            Console.WriteLine("│         │");
            Console.CursorLeft = indentSize;
            Console.WriteLine("│         │");
            Console.CursorLeft = indentSize;
            Console.WriteLine($"│      {r} │");
            Console.CursorLeft = indentSize;
            Console.WriteLine("└─────────┘");
        }
        string RankToString() //Метод преобразования достоинтсва карты в строку
        {
            switch (rank)
            {
                case Rank._6:
                case Rank._7:
                case Rank._8:
                case Rank._9:
                    return $"{(int)rank} ";
                case Rank._10:
                    return $"{(int)rank}";
                case Rank.Jack:
                    return "J ";
                case Rank.Queen:
                    return "Q ";
                case Rank.King:
                    return "K ";
                case Rank.Ace:
                    return "A ";
                default:
                    return null;
            }
        }
    }
}
